ios/RNCAsyncStorageDelegate.h

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^RNCAsyncStorageCompletion)(NSError *_Nullable error);
+typedef void (^RNCAsyncStorageResultCallback)(NSArray<id<NSObject>> * valuesOrErrors);
+
+@protocol RNCAsyncStorageDelegate <NSObject>
```

Comment:
@tido64 Thanks for work. I think it's necessary to support custom storage for users, currently if we set delegate, all operations would fall back to delegate. Ideally, I may like add an option to enable/disable built-in storage, the use case is, for example, if user only wants to observer the behavior of operations, not want to implement custom storage. @Krizzu What's your opinion? 🤔 .

Reply:
FWIW, it should be trivial to add a property on the delegate whether to use it as a passthrough. If set to `YES`, we don't return and just continue as usual.

Reply:
@tido64 	Sounds good to me. This would make AS more flexible for brownfield apps. @tido64 	What do you think?

Reply:
I think this makes a lot of sense. There will be apps that want to monitor disk usage etc. but don't want to handle the storage themselves. I've just pushed the changes, let me know what you think. I'll work on docs/tests when I have time next.